autoload/lightline/gitdiff.vim

```diff
@@ -34,12 +34,23 @@ endfunction
 function! lightline#gitdiff#write_calculation_to_cache(buffer, soft) abort
   if a:soft && has_key(g:lightline#gitdiff#cache, a:buffer)
     " b/c there is something in the cache already
-    return 
+    return
   endif
 
-  let l:Calculation = get(g:, 'lightline#gitdiff#algorithm',
-        \ { buffer -> lightline#gitdiff#algorithms#word_diff_porcelain#calculate(buffer) })
-  let g:lightline#gitdiff#cache[a:buffer] = l:Calculation(a:buffer)
+  let l:indicator_values = get(g:, 'lightline#gitdiff#algorithm',
+      \ { buffer -> lightline#gitdiff#algorithms#word_diff_porcelain#calculate(buffer) })(a:buffer)
+
+  " If the user doesn't want to show empty indicators,
+  "     then remove the empty indicators returned from the algorithm
+  if !get(g:, 'lightline#gitdiff#show_empty_indicators', 0)
+    for key in keys(l:indicator_values)
+      if l:indicator_values[key] == 0
+        unlet l:indicator_values[key]
+      endif
+    endfor
+  endif
```

Comment:
Wouldn't this keep the indicators if the user hadn't set the `show_empty_indicators` variable (or set it to `0`) instead of removing them?

My thought was to have the algorithm return the "full" git state and have `write_calculation_to_cache` figure out which indicators to show.

Reply:
This is `write_calculation_to_cache()` isn't it? 🙂

Anyway, you're totally right, I mixed up the boolean. But the version you provided should be more perfomant because checking for the configuration value is only done once.